### PR TITLE
[3/5] Added async example that interacts with ERC20 token

### PR DESCRIPTION
**Pull request by nlordell**
Also included some minor fixes that the example brought to light like:
- parsing hex string was failing because of leading "0x"
- truffle seems to differentiate between `to: null` and `to: 0x00...0`

### CODE REVIEW

File: src/transaction.rs [hunk 1]
```diff
@@ -33,7 +33,7 @@ pub struct TransactionBuilder<T: Transport> {
     /// to locally signing on the node with the default acount.
     pub from: Option<Account>,
     /// The receiver of the transaction.
-    pub to: Address,
+    pub to: Option<Address>,
```
Comment (fleupold) [inline src/transaction.rs:36]: Why this change? What happens if to is None? Is that valid?

File: common/src/truffle/bytecode.rs [hunk 2]
```diff
@@ -83,7 +83,7 @@ impl Bytecode {
     pub fn into_bytes(self) -> Result<Bytes, LinkError> {
         match self.undefined_libraries().next() {
             Some(library) => Err(LinkError::UndefinedLibrary(library.to_string())),
-            None => Ok(Bytes(hex::decode(&self.0).expect("valid hex"))),
+            None => Ok(Bytes(hex::decode(&self.0[2..]).expect("valid hex"))),
```
Comment (fleupold) [inline common/src/truffle/bytecode.rs:86]: How did this work before?

File: Cargo.toml [hunk 3]
```diff
@@ -38,3 +38,7 @@ serde_json = "1.0"
 thiserror = "1.0"
 tiny-keccak = "1.5"
 web3 = "0.8"
+
+[dev-dependencies]
+anyhow = "1.0"
```
Comment (fleupold) [inline Cargo.toml:43]: How is this used?

File: Cargo.toml [hunk 4]
```diff
@@ -38,3 +38,7 @@ serde_json = "1.0"
...
+rustversion = "1.0"
```
Comment (fleupold) [inline Cargo.toml:44]: Shouldn't this only be used by the example feature and thus be specified in the feature section?

**Review by fleupold — APPROVED** [review]
<a href="https://gitme.me/image?url=https%3A%2F%2Fmedia1.giphy.com%2Fmedia%2FCjmvTCZf2U3p09Cn0h%2Fgiphy-downsized-medium.gif&token=lets_go" data-gitmeme-token="lets_go"></a>

File: src/transaction.rs (hunk 1 above)
Comment (nlordell) [reply src/transaction.rs:36]: When doing JSON RPC calls to Ganache, sending request `{ to: "0x0..0" }` makes contract creation fail while `{ to: null }` works, which is what `None` lets us do. :face_with_head_bandage:

File: common/src/truffle/bytecode.rs (hunk 2 above)
Comment (nlordell) [reply common/src/truffle/bytecode.rs:86]: It didn't... I didn't have deploy unit tests so I never caught it until I fixed it for the async example. Here is the true culprit:
https://github.com/gnosis/ethcontract-rs/blob/9fd4f359c110928774ad968620d0e27641f07651/src/contract/deploy.rs#L302

File: Cargo.toml (hunk 3 above)
Comment (nlordell) [reply Cargo.toml:43]: `dev-dependencies` are dependencies that only get used for ~~build scripts~~, testing, examples and benchmarking. I think doc tests also uses it.

**edit**: correction

File: Cargo.toml (hunk 4 above)
Comment (nlordell) [reply Cargo.toml:44]: `example` feature is for generating example contract code to get included in the documentation. I think I might rename the feature so it doesn't get confused with `cargo run --example`

File: common/src/truffle/bytecode.rs (hunk 2 above)
Comment (nlordell) [reply common/src/truffle/bytecode.rs:86]: See #2

File: Cargo.toml (hunk 4 above)
Comment (nlordell) [reply Cargo.toml:44]: Created issue #27

File: Cargo.toml (hunk 3 above)
Comment (fleupold) [reply Cargo.toml:43]: Right, I was wondering who uses anyhow, since I could not find it in the code.

File: Cargo.toml (hunk 4 above)
Comment (fleupold) [reply Cargo.toml:44]: My question was more since rustversion is only used by the example feature, can we make this dependency be only downloaded and built when compiled with example feature enabled.

Comment (nlordell) [reply Cargo.toml:44]: Its used regardless of `example` feature as our `async` example gets built with or without that feature.
The `example` feature just includes or excludes the `ethcontract::example` module.

File: Cargo.toml (hunk 3 above)
Comment (nlordell) [reply Cargo.toml:43]: Oh! `async.rs` used to use it. I must have forgotten to remove the dependency
